### PR TITLE
Implemented normalizing optionals on input objects

### PR DESCRIPTION
**Pull request by xperiandri**
@valbers I've implemented minimal tests. Do a brief check, please

### CODE REVIEW

File: Packages.props [hunk 1]
```diff
@@ -73,5 +73,6 @@
     <PackageReference Update="Microsoft.Diagnostics.Tracing.TraceEvent" Version="3.*" />
     <PackageReference Update="Microsoft.NETCore.Platforms" Version="6.0.*" />
     <PackageReference Update="Newtonsoft.Json" Version="13.*" />
+    <PackageReference Update="Validus" Version="4.1.*" />
```
Comment (valbers) [inline Packages.props:76]: what about `FsToolkit.ErrorHandling`?

File: src/FSharp.Data.GraphQL.Server/ReflectionHelper.fs [hunk 2]
```diff
@@ -130,10 +130,13 @@ module internal Gen =
 
 module internal ReflectionHelper =
 
+    let [<Literal>] OptionTypeName = "Microsoft.FSharp.Core.FSharpOption`1"
```
Comment (valbers) [inline src/FSharp.Data.GraphQL.Server/ReflectionHelper.fs:133]: Good 👍

File: src/FSharp.Data.GraphQL.Server/Values.fs [hunk 3]
```diff
@@ -312,6 +322,16 @@ let rec internal coerceVariableValue
             | Ok null when isNullable -> Ok null
             // TODO: Capture position in the JSON document
             | Ok null -> createNullError originalTypeDef
+            | Ok value when not isNullable ->
+                let ``type`` = value.GetType()
```
Comment (valbers) [inline src/FSharp.Data.GraphQL.Server/Values.fs:326]: it's probably a matter of taste, but I find the name
```
``type``
```
rather cumbersome. The backticks trick is a genius move for (some, mostly unit-test) methods, but as a variable name, as I said, it's IMHO cumbersome.
I guess `type_` should also do the trick.
Just my two cents; do as you wish 🙂

**Review by valbers — APPROVED** [review]

File: src/FSharp.Data.GraphQL.Server/Values.fs (hunk 3 above)
Comment (xperiandri) [reply src/FSharp.Data.GraphQL.Server/Values.fs:326]: In the other code bases I saw `ty` and `typ` but it looks bad as for me 🙂

File: Packages.props (hunk 1 above)
Comment (xperiandri) [reply Packages.props:76]: It is referenced transitively implicitly. Do you think we need to reference it explicitly?

**Pull request merged**